src/pages/index.js

```diff
@@ -54,26 +124,77 @@ addButton.addEventListener("click", () => {
   addNewCardFormValidator.disableSubmitButton();
   addNewCardModal.open();
 });
+updateAvatarIcon.addEventListener("click", () => {
+  updateAvatarFormValidator.disableSubmitButton();
+  updateAvatarModal.open();
+});
 
 export const userInfo = new UserInfo({
   userNameSelector: ".profile__heading",
   userOccupationSelector: ".profile__sub-heading",
+  profileAvatarSelector: ".profile__avatar",
 });
 
 export function updateProfile(profileObject) {
   const newName = profileObject["input-name"];
   const newOccupation = profileObject["input-description"];
+  console.log(profileObject);
+  profileSubmitButton.textContent = "Saving...";
+
+  api
+    .updateProfile({
+      name: newName,
+      occupation: newOccupation,
+    })
+    .finally(() => {
+      profileSubmitButton.textContent = "Save";
+    });
   userInfo.setUserInfo(newName, newOccupation);
   editProfileFormValidator.disableSubmitButton();
   editProfileModal.close();
```

Comment:
You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well

**Please, correct it everywhere**